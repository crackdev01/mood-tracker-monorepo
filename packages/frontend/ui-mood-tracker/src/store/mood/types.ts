// Define Mood Store State.
export interface MoodState {
  mood: [];
}

// Define Mood Store actions.
export enum MoodActions {
  FETCH_MOODS = 'FETCH_MOODS',
  RENDER_MOODS = 'RENDER_MOODS',
}
