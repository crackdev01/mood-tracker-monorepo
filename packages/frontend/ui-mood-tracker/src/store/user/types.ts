export interface UserRequest {
  username: string;
  password: string;
}

export interface Jwt {
  exp: number;
  iat: number;
  username: string;
  uuid: string;
}

export interface Coordinates {
  latitude: number;
  longitude: number;
}

export interface User {
  accessToken: string;
  decodedAccessToken: Jwt | null;
  // We set the coordinate values if user has provided permission.
  // Else we set to false.
  location?: Coordinates | string;
}

// Define User Store State.
export interface UserState {
  user: User;
}

// Define User Store actions.
export enum UserActions {
  FETCH_USER = 'FETCH_USER',
  RENDER_USER = 'RENDER_USER',
  LOGOUT_USER = 'LOGOUT_USER',
  SET_LOCATION_SUCCESS = 'SET_LOCATION_SUCCESS',
  SET_LOCATION_DENIED = 'SET_LOCATION_DENIED',
  SET_LOCATION_ERROR = 'SET_LOCATION_ERROR',
}
