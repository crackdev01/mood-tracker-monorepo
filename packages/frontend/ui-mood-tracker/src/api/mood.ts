import { authMoodTrackerApi } from './core';

// FIXME: update types.
export async function getMoodEntries(): Promise<any> {
  return await authMoodTrackerApi.get('/mood/entries');
}

export async function postMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.post('/mood/entry', payload);
}

export async function updateMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.put('/mood/entry', payload);
}

export async function deleteMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.delete(`/mood/entry/${payload.id}`);
}
