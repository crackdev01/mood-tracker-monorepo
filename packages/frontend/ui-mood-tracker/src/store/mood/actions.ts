import { MoodActions } from './types';

export const fetchMoodEntries = () => {
  return {
    type: MoodActions.FETCH_MOODS,
  };
};

export const addMoodEntry = () => {
  return {
    type: MoodActions.ADD_MOOD,
  };
};

export const editMoodEntry = () => {
  return {
    type: MoodActions.EDIT_MOOD,
  };
};

export const deleteMoodEntry = () => {
  return {
    type: MoodActions.DELETE_MOOD,
  };
};

// FIXME: fix type.
export const renderMoodEntries = (mood: any) => {
  return {
    type: MoodActions.RENDER_MOODS,
    mood,
  };
};
