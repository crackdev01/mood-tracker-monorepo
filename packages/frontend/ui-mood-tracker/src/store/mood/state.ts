import { MoodState } from './types';

export const moodState: MoodState = {
  mood: [],
};
