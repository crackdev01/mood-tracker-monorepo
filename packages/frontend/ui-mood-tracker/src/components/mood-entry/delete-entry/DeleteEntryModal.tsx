import React from 'react';
import { useDispatch } from 'react-redux';
import dayjs from 'dayjs';
import { Button, Header, Icon, Modal } from 'semantic-ui-react';
import { useTranslation } from 'react-i18next';

import { MoodActions } from '../../../store/mood/types';

import './_delete-entry.scss';

const DeleteEntryModal = (props: { displayModal: boolean; closeModal: any; mood: any }) => {
  const { displayModal, closeModal, mood } = props;
  const dispatch = useDispatch();
  const { t } = useTranslation(['MoodEntry']);

  const deleteMoodEntry = () => {
    const payload = {
      id: props.mood.mood_id,
    };
    dispatch({
      type: MoodActions.DELETE_MOOD,
      payload,
    });
    closeModal();
  };

  return (
    <Modal className="delete-entry" closeIcon open={displayModal} onClose={closeModal} size="tiny">
      <Header icon="delete" content={t('header')} />
      <Modal.Content className="delete-entry__content">
        <h4>{t('deleteEntry.header')}</h4>
        <div>
          <b>{t('table.headers.status')}: </b>
          {mood.mood_status}
        </div>
        <div>
          <b>{t('table.headers.intensity')}: </b>
          {mood.mood_intensity}
        </div>
        <div>
          <b>{t('table.headers.enteredAt')}: </b>
          {dayjs(mood.mood_enteredAt).format('MMM D, YYYY h:mm A')}
        </div>
      </Modal.Content>
      <Modal.Actions className="delete-entry__actions">
        <Button className="delete-entry__actions__yes" color="green" onClick={deleteMoodEntry}>
          <Icon name="checkmark" /> {t('buttons.yes')}
        </Button>
        <Button className="delete-entry__actions__no" onClick={closeModal}>
          <Icon name="remove" /> {t('buttons.no')}
        </Button>
      </Modal.Actions>
    </Modal>
  );
};

export default DeleteEntryModal;
