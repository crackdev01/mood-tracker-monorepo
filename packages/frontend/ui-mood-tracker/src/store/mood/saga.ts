import { all, call, put, takeEvery, fork } from 'redux-saga/effects';

import { MoodActions } from './types';
import { getMoodEntriesApi, postMoodEntry, updateMoodEntry, deleteMoodEntry } from '../../api/mood';

// FIXME: update types in file.

// Fetch mood entries.
export function* getMoodEntriesCall() {
  const { data } = yield call(getMoodEntriesApi);
  yield put({ type: MoodActions.RENDER_MOODS, data });
}

export function* getMoodEntries() {
  yield takeEvery(MoodActions.FETCH_MOODS, getMoodEntriesCall);
}

// Add mood entry.
export function* addMoodEntryCall(action: any) {
  yield call(postMoodEntry, action.payload);
  yield call(getMoodEntriesCall);
}

export function* addMoodEntry() {
  yield takeEvery(MoodActions.ADD_MOOD, addMoodEntryCall);
}

// Update mood entry.
export function* editMoodEntryCall(action: any) {
  yield call(updateMoodEntry, action.payload);
  yield call(getMoodEntries);
}

export function* editMoodEntry() {
  yield takeEvery(MoodActions.EDIT_MOOD, editMoodEntryCall);
}

// Delete mood entry.
export function* removeMoodEntryCall(action: any) {
  yield call(deleteMoodEntry, action.payload);
  yield call(getMoodEntries);
}

export function* removeMoodEntry() {
  yield takeEvery(MoodActions.DELETE_MOOD, removeMoodEntryCall);
}

export function* moodSaga() {
  yield all([fork(getMoodEntries), fork(addMoodEntry), fork(editMoodEntry), fork(removeMoodEntry)]);
}

export default moodSaga;
