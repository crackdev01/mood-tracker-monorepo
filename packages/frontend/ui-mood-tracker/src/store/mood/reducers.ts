import { moodState } from './state';
import { MoodActions } from './types';

export const mood = (state = moodState, action: any) => {
  switch (action.type) {
    case MoodActions.FETCH_MOODS:
      return {
        ...state,
        mood: action.mood,
      };
    default:
      return state;
  }
};
