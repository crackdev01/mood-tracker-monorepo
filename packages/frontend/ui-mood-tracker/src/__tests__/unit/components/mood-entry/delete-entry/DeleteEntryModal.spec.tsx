import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DeleteEntryModal from '../../../../../components/mood-entry/delete-entry/DeleteEntryModal';
import { mockMoods } from '../../../../fixtures';
import { MoodActions } from '../../../../../store/mood/types';

Enzyme.configure({ adapter: new Adapter() });

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('DeleteEntryModal', () => {
  const closeModalMock = jest.fn();
  const props = {
    displayModal: true,
    closeModal: closeModalMock,
    mood: mockMoods[0],
  };

  test('renders', () => {
    mockDispatch.mockReturnValueOnce({});
    const w = shallow(<DeleteEntryModal {...props} />);
    expect(w.debug()).toMatchSnapshot();
  });

  test('triggers deleteMoodEntry', () => {
    mockDispatch.mockReturnValueOnce({});
    const w = shallow(<DeleteEntryModal {...props} />);
    w.find('Button').at(1).simulate('click');
    expect(mockDispatch).toHaveBeenCalledWith({
      type: MoodActions.DELETE_MOOD,
      payload: { id: mockMoods[0].mood_id },
    });
    expect(closeModalMock).toHaveBeenCalledTimes(1);
  });
});
