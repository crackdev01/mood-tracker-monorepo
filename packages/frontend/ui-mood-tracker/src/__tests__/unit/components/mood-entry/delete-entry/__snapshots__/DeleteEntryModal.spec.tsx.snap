// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteEntryModal renders 1`] = `
"<Modal className=\\"delete-entry\\" closeIcon={true} open={true} onClose={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} size=\\"tiny\\" centered={true} dimmer={true} closeOnDimmerClick={true} closeOnDocumentClick={false} eventPool=\\"Modal\\">
  <Header icon=\\"delete\\" content=\\"header\\" />
  <ModalContent className=\\"delete-entry__content\\">
    <h4>
      deleteEntry.header
    </h4>
    <div>
      <b>
        table.headers.status
        : 
      </b>
      relaxed
    </div>
    <div>
      <b>
        table.headers.intensity
        : 
      </b>
      0
    </div>
    <div>
      <b>
        table.headers.enteredAt
        : 
      </b>
      Jun 20, 2021 12:00 AM
    </div>
  </ModalContent>
  <ModalActions className=\\"delete-entry__actions\\">
    <Button className=\\"delete-entry__actions__yes\\" color=\\"green\\" onClick={[Function: deleteMoodEntry]} as=\\"button\\">
      <Icon name=\\"checkmark\\" as=\\"i\\" />
       
      buttons.yes
    </Button>
    <Button className=\\"delete-entry__actions__no\\" onClick={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} as=\\"button\\">
      <Icon name=\\"remove\\" as=\\"i\\" />
       
      buttons.no
    </Button>
  </ModalActions>
</Modal>"
`;
